TODO1/encapsulating-style-and-structure-with-shadow-dom.md

```diff
@@ -154,19 +154,19 @@ A simple example would look like this:
 </div>
 ```
 
-A given shadow root can have any number of slot elements, which can be distinguished with a `name` attribute. The first slot inside of the shadow root without a name, will be the default slot and all content not otherwise assigned will flow inside that node. Our dialog really needs two slots: a heading and some content (which we’ll make default).
+一个给定的 shadow root 可以拥有任意数量的插槽元素，可以用 `name` 属性来区分。Shadow root 中没有名称的第一个插槽将是默认插槽，未分配的所有内容将在该节点内按文档流（从左到右，从上到下）显示。我们的对话框确实需要两个插槽：标题和一些内容（我们将设置为默认插槽）。
```

Comment:
我觉得这个slot 别翻译了？ 因为还蛮突然的。如果真的不知道slot 这个文章其实也没法读。。。我觉得去平衡 翻译和不翻译真的很难。。。。

然后这边 这个codepen 我不知道你看没看。 我把代码贴出来
```
<one-dialog>
  <span slot="heading">Hello world</span>
  <div>
    <p>Lorem ipsum dolor amet tilde bicycle rights affogato brooklyn. Whatever lomo subway tile sriracha gastropub edison bulb shabby chic tumeric meditation mustache raw denim.</p>
    
    <p>reegan ugh bespoke you probably haven't heard of them godard crucifix pabst. Selvage biodiesel vice copper mug lumbersexual cred plaid. Skateboard pitchfork listicle fashion axe. Chillwave viral butcher vegan wolf.</p>
  </div>
</one-dialog>

<button id="launch-dialog">Launch dialog</button>
```

这里的default slot 是   `<div>... </div>`  heading 这个有那么啦

然后我在想

 The first slot inside of the shadow root without a name, will be the default slot and all content not otherwise assigned will flow inside that node. 
这里的 not assigned 是不是就是其他没有命名的 slot

Reply:
嗯有可能，但我们的意思是一致的，都是除了插槽以外的其他内容